packages/@stylexjs/eslint-plugin/src/stylex-no-unused.js

```diff
@@ -132,48 +158,17 @@ const stylexNoUnused = {
       };
     }
 
-    function parseStylexImportStyle(node: Node) {
-      // identify stylex import
-      if (
-        node.source?.value === '@stylexjs/stylex' &&
-        // $FlowFixMe[prop-missing]
-        node.importKind === 'value' &&
-        node.specifiers &&
-        node.specifiers.length > 0
-      ) {
-        // extract stylex import pattern
-        node.specifiers.forEach((specifier) => {
-          const specifierType = specifier.type;
-          if (specifierType === 'ImportNamespaceSpecifier') {
-            // import * as stylex from '@stylexjs/stylex';
-            stylexImportObject = specifier.local.name;
-            stylexImportProperty = 'create';
-          } else if (specifierType === 'ImportDefaultSpecifier') {
-            if (specifier.local.name === 'stylex') {
-              // import stylex from '@stylexjs/stylex';
-              stylexImportObject = 'stylex';
-              stylexImportProperty = 'create';
-            }
-          } else if (specifierType === 'ImportSpecifier') {
-            if (specifier.imported?.name === 'create') {
-              // import {create} from '@stylexjs/stylex'  OR   import {create as c} from '@stylexjs/stylex'
-              stylexImportObject = '';
-              stylexImportProperty = specifier.local.name;
-            }
-          }
-        });
-      }
-    }
-
     return {
       Program(node: Program) {
-        // detect stylex import style, which then decides which variables are stylex styles
+        // Process imports first
         node.body
-          .map((node) => (node.type === 'ImportDeclaration' ? node : null))
-          .filter(Boolean)
-          .forEach(parseStylexImportStyle);
-        // stylex.create can only be singular variable declarations at the root
-        // of the file so we can look directly on Program and populate our set.
+          .filter(
```

Comment:
we need to ensure collect import statements before variable detection happens, so this can't happen in `ImportDeclaration` and needs to be done in `Program`